packages/dear-imgui/cpp/src/widget/slider.cpp

```diff
@@ -0,0 +1,34 @@
+#include <string>
+
+#include "widget/slider.h"
+#include "widget/styled_widget.h"
+
+void Slider::Render(ReactImgui* view) {
+    ImGui::PushID(m_id);
+    if (m_type == "angle") {
+        if (ImGui::SliderAngle(m_label.c_str(), &m_value, m_min, m_max, "%.0f")) {
+            view->m_onNumericValueChange(m_id, m_value);
+        }
+    } else {
+        if (ImGui::SliderFloat(m_label.c_str(), &m_value, m_min, m_max, "%.0f")) {
+            view->m_onNumericValueChange(m_id, m_value);
+        }
+    }
+    ImGui::PopID();
+};
+
+void Slider::Patch(const json& widgetPatchDef, ReactImgui* view) {
+    if (widgetPatchDef.is_object()) {
+        StyledWidget::Patch(widgetPatchDef, view);
+
+        if (widgetPatchDef.contains("label") && widgetPatchDef["label"].is_string()) {
+            m_label = widgetPatchDef["label"].template get<std::string>();
+        }
+        if (widgetPatchDef.contains("min") && widgetPatchDef["min"].is_number()) {
+            m_min = widgetPatchDef["min"].template get<float>();
+        }
+        if (widgetPatchDef.contains("max") && widgetPatchDef["max"].is_number()) {
+            m_max = widgetPatchDef["max"].template get<float>();
+        }
+    }
```

Comment:
Neither m_sliderType nor value can be directed via Patch(). I'll add an imperative handle so that it will be possible to update the value programmatically from within the React layer. I am not convinced that the slider type should be editable.